File: samples/capture/python/cap.py
import cv2

width = 640
height = 480
cap = cv2.VideoCapture()
cap.set(cv2.CAP_PROP_FRAME_WIDTH, width)
cap.set(cv2.CAP_PROP_FRAME_HEIGHT, height)
cap.open(0)
while True:
    ret, frame = cap.read()
    cv2.imshow("capture", frame)
    key = cv2.waitKey(10)
    if key == 27:
        break

cap.release()
cv2.destroyAllWindows()

File: samples/capture/cpp/src/cap.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

int width = 640;
int height = 480;

int main(int argc, char *argv[])
{
    cv::VideoCapture cap;
    cap.set(cv::CAP_PROP_FRAME_WIDTH, width);
    cap.set(cv::CAP_PROP_FRAME_HEIGHT, height);
    cap.open(0);
    cv::Mat frame;
    while (true)
    {
        cap >> frame;
        if (frame.empty())
        {
            break;
        }
        cv::imshow("capture", frame);
        if (cv::waitKey(10) == 27)
        {
            break;
        }
    }
    cap.release();
    cv::destroyAllWindows();
    return 0;
}
